cmd/cmp/main_test.go

```diff
@@ -0,0 +1,33 @@
+package cmp
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+func TestRun(t *testing.T) {
+
+	FileIdRegexStr = `.*(query_\d{2})(?:_c0)?\.output`
+
+	expectedReturns := []int{0, 1, 1, 1, 0, 0}
+	for i := 0; i <= 5; i++ {
+		outputDir, err := os.MkdirTemp("", fmt.Sprintf("test_%d_output_", i))
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.RemoveAll(outputDir)
```

Comment:
Using `defer` in the loop iteration is not a idiomatic way and may even cause resource leak.
You can refer to this answer which I think is a good explanation. It also has an example about how to handle this. https://stackoverflow.com/questions/45617758/proper-way-to-release-resources-with-defer-in-a-loop/45620423#45620423

Reply:
The StackOverflow solution to this is to wrap each loop iteration in a anonymous function. However, in your case, you can simply put `os.RemoveAll(outputDir)` at the end of each iteration because there is no early exit in your code. Please let me know if I need to have a call with you to further explain this.

Reply:
Thanks for the explanation! I think I understand what you mean, I can see why there is an issue with this. The defer function only executes when the surrounding function finishes, not when the loop ends. What you want is just for me to call the os.RemoveAll() function since the cleanup needs to be executed at the end of each iteration.